app/scripts/lib/transform.js

```diff
@@ -40,6 +40,36 @@ define(function (require, exports, module) {
       }
 
       return result.value;
+    },
+    /**
+     * Simplifies user agent operating system names (50+) to generic popular names (~6)
+     *
+     * @param {String} os Operating System name from
+     * @returns {String} generic operating system name
+     */
+    toGenericOSName (os) {
```

Comment:
I just had an stemming from the discussion we had IRL, this function seems a little misplaced here. I think this can be moved to user-agent.js, as a "static" function that lives off of the UAP constructor.

Example:

```js
const UserAgent = require('lib/user-agent');
...
const genericOSName = UserAgent.toGenericOSName(osName);
```